src/model/utils/XrefUtils.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  databaseCategoryToDatabases,
+  entrySectionToDatabaseCategories,
+} from '../../data/database';
+import EntrySectionType from '../types/EntrySectionType';
+import { Databases, DatabaseCategory } from '../types/DatabaseTypes';
+import { Property } from '../types/modelTypes';
+
+export type Xref = {
+  databaseType?: Databases;
+  id?: string;
+  properties?: [Property];
+  isoformId?: string;
+};
+
+export type XrefsGoupedByDatabase = {
+  database: Databases;
+  xrefs: Xref[];
+};
+
+export type XrefUIModel = {
+  category: DatabaseCategory;
+  databases: XrefsGoupedByDatabase[];
+};
+
+export const getXrefsForSection = (
+  xrefs: Xref[],
+  section: EntrySectionType
+): XrefUIModel[] | null => {
+  const xrefCategories: XrefUIModel[] = [];
+  // Get the categories relevant to the given entry section
+  const sectionCategories = entrySectionToDatabaseCategories.get(section);
+  if (!sectionCategories) {
+    return null;
+  }
+  sectionCategories.forEach(category => {
+    // Get the database relevant to the given category
+    const databases = databaseCategoryToDatabases.get(category);
+    const categoryDatabases: XrefsGoupedByDatabase[] = [];
+    if (databases) {
+      databases.forEach(database => {
+        // Filter the xref data to only return the ones for
+        // the given database
+        const databaseXrefs = xrefs.filter(
```

Comment:
This refactoring is more intuitive than the previous version. However, it will be slightly more CPU intensive as all of the  `databaseCrossReferences` will be filtered `|sectionCategories| * |databases for category|` times. For example, for P05067 there are 14 + 34 + 19 = 67 `sectionCategories`/`databases for category` loops and `databaseCrossReferences` has 782 entries which will mean a total of 52,394 equality checks: `xref.databaseType === database`. We haven't finalized performance testing but this block of code may be something to consider in the future if we need to shave off some ms off the loading time.

Reply:
Very good point. Might be worth looking at creating maps at load time to speed up the data retrieval. For now I've added a comment `// TODO Performance improvement`, would be good to add these wherever we spot a potential bottleneck on which we can improve later.